File: backend/src/main/java/com/example/Flicktionary/domain/actor/dto/ActorDto.java
package com.example.Flicktionary.domain.actor.dto;

import com.example.Flicktionary.domain.actor.entity.Actor;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.lang.NonNull;

@Getter
@AllArgsConstructor
public class ActorDto {
    @NonNull
    private final Long id;
    @NonNull
    private final String name;
    private final String profilePath;

    public ActorDto(Actor actor) {
        this(actor.getId(), actor.getName(), actor.getProfilePath());
    }
}

File: backend/src/main/java/com/example/Flicktionary/domain/actor/dto/ActorDto.kt
package com.example.Flicktionary.domain.actor.dto

import com.example.Flicktionary.domain.actor.entity.Actor

data class ActorDto(
    val id: Long,
    val name: String,
    val profilePath: String?
) {
    constructor(actor: Actor) : this(actor.id, actor.name, actor.profilePath)
}